docs/DeveloperGuide.md

```diff
@@ -1,34 +1,965 @@
+`storage`. These 3 parameters aid with printing information to the user, making modifications to the data and saving the
+data. Each subclass of `Command` overrides `execute()` and implements their own methods to execute the command. Each 
+subclass also has a unique constructor signature as each subclass requires different parameters to execute.
+
+#### Implementation
+
+The implementation for executing every command differs, and the implementation details of each of them will be further 
+elaborated below.
+
+#### 4.3.1 Add Command
+
+The add command allows a user to add a new module to the list of existing modules. The module name is part of the 
+parameters and is extracted directly from user input while the various information required to add a new module are 
+included in the flags of the user input. There are 3 compulsory flags and 1 optional flag for adding a module:
+
+1. module code
+    - `-c <String>`
+1. module credits
+    - `-mc <double>`
+1. module type
+    - `-t <String>`
+1. (Optional) prerequisite modules
+    - `-p [>String>, ...]`
+
+> ℹ️ **Note:** The order of flags in user input does not matter.
+
+The sequence diagram below shows the execution of add command in action:
+
+![archi](./images/AddCommandSequenceDiagram.png)
+
+<sup>***Figure 4.3.1.1** Sequence diagram of `AddCommand` in execution.*</sup>
+
+#### 4.3.2 Delete Command
+
+The delete command allows for deletion of module from the module list, identified by the module code. There are no flags
+involved for deleting a module.
+
+> ℹ️ **Note:** Users cannot delete modules which are prerequisites for other modules.
+
+![archi](./images/DeleteCommandSequenceDiagram.png)
+<sup>***Figure 4.3.2.1** Sequence diagram of `Delete` class with "delete CS1010" as user input*</sup>
+
+#### 4.3.3 Update Command
+
+The update commands allows modifications to the existing modules, identified by the module code. 
+The information that can be updated include module name, credits, prerequisites and grades (if the module is 
+marked as done). 
+
+The update function is executed if the user decides to edit some information regarding a module in the module
+list. The various information requested to update would be identified with their flags: 
+1. name 
+   - `-n <String>`
+1. module credits
+   - `-mc <int>`
+1. prerequisites
+   - `-p [<String>, ...]`
+1. grade
+   - `-g <String>`
+
+
+> ℹ️ **Note:** The code and type of modules <b>cannot be modified</b> as they are identifiers of the modules.
+
+> ℹ️ **Note:** <b>Multiple module information</b> can be updated in a single command
+
+> ℹ️ **Note:** The command <b>will not update grades</b> if the module requested has not been completed. The rest of 
+the information parsed in the command (if any) will be updated. 
+
+
+#### Details
+
+The following is the UML diagrams for update command consisting of updating each flag.
+
+![archi](./images/updateClassDiagram.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/10681776/113077325-82af0f00-9203-11eb-85a4-58affa5c7375.png)

Shouldn't these roles and navigability numbers be at the _right_ side (marked by arrow) instead?